src/providers/wms/qgswmssourceselect.cpp

```diff
@@ -1020,6 +1023,39 @@ void QgsWMSSourceSelect::collectSelectedLayers( QStringList &layers, QStringList
   }
 }
 
+void QgsWMSSourceSelect::collectDimensions( QStringList &layers, QgsDataSourceUri &uri )
+{
+  for ( const QgsWmsLayerProperty layerProperty : mLayerProperties )
+  {
+    if ( layerProperty.name == layers.join( QStringLiteral( "," ) ) )
```

Comment:
```suggestion
    if ( layerProperty.name == layers.join( ',' ) )
```

The character based overrides for string operations are MUCH MUCH faster then the QString versions